### PR TITLE
Remove runcfanotify package

### LINKED ISSUE #1831: Deprecate and delete runc-fanotify in favour of container-hook
runc-fanotify and container-hook are similar and share some code but have subtle differences. When making a change in one, we often need to update the other one and we risk forgetting.

I suggest to wait until September and then delete runc-fanotify if we have not found any issue with container-hook in various environments.

### PR DESCRIPTION
**Pull request by alban**
It has been replaced by the container-hook package.

runcfanotify only works when Inspektor Gadget runs in the host pid namespace. But this is not the case since June 2023.

This also removes the option "kubectl-gadget deploy --legacy-host-pid".

See also #1709

Fixes #1831

### CODE REVIEW

File: examples/runc-hook/README.md
Comment (flyth) [inline examples/runc-hook/README.md:1]: Maybe rename `examples/runc-hook` to `examples/container-hook`?

**Review by flyth — APPROVED** [review]
LGTM from code inspection. Maybe just rename the example as well.

File: examples/runc-hook/README.md
Comment (alban) [reply examples/runc-hook/README.md:1]: examples/container-hook already exists though... but it is less complete.
Should I delete examples/container-hook and then do the rename?

Comment (flyth) [reply examples/runc-hook/README.md:1]: Ah - I'd choose to keep the more complete solution, then. But I'm fine with whatever you decide.

**Comment by alban** [comment]
Examples updated. I also pushed to branch `citest/alban_delete_runc_fanotify`.

**Comment by alban** [comment]
Tests with `citest/alban_delete_runc_fanotify` passed fine, after 1 retry:
https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/12029998050

**Comment by alban** [comment]
I checked that the runcfanotify removal is fine for Kubescape. Kubescape does not use runcfanotify since December 2023, see:
- https://github.com/kubescape/node-agent/blob/v0.2.185/pkg/containerwatcher/v1/container_watcher_private.go#L117-L118
- https://github.com/kubescape/node-agent/pull/153 ("The container collection options have been updated to use containercollection.WithContainerFanotifyEbpf instead of containercollection.WithRuncFanotify")

**Pull request merged**